### PR TITLE
Aduric/botify

### PR DESCRIPTION
**Pull request by aduric**

### CODE REVIEW

**Comment by aduric** [comment]
Initial slack-rs handler. Pierre listens to all channel events and outputs to stdout.

File: bot/src/main.rs [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+    let filepath = format!("{}/.pierre_config", home_dir.to_string_lossy());
+    let mut config = Config::load(&filepath).expect(&format!("Could not load config at {}", filepath));
+
+    let mut handler = EventHandler::new(config.slack.user);
+    let mut cli = slack_api::RtmClient::new(config.slack.token.as_str());
+    let r = cli.login_and_run::<EventHandler>(&mut handler);
+    match r {
+        Ok(_) => {}
+        Err(err) => panic!("Error: {}", err),
+    }
```
Comment (mthjones) [inline bot/src/main.rs]: Consider either using `expect(...)` here or using `if let Err(err) = r { ... }`

File: bot/src/main.rs [hunk 2]
```diff
@@ -1,3 +1,23 @@
+extern crate slack as slack_api;
+extern crate pierre;
+
+mod slack;
```
Comment (mthjones) [inline bot/src/main.rs]: Consider renaming this module to `event_handler` or similar.

File: bot/src/slack/mod.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+use slack_api::{self, Event, Message};
```
Comment (mthjones) [inline bot/src/slack/mod.rs]: This `mod.rs` file could just be `slack.rs` in the top-level directory.

File: bot/src/main.rs [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
-    println!("Hello, world!");
+    
+    let home_dir = std::env::home_dir().expect("Could not find home directory to place config");
+    let filepath = format!("{}/.pierre_config", home_dir.to_string_lossy());
+    let mut config = Config::load(&filepath).expect(&format!("Could not load config at {}", filepath));
+
+    let mut handler = EventHandler::new(config.slack.user);
+    let mut cli = slack_api::RtmClient::new(config.slack.token.as_str());
+    let r = cli.login_and_run::<EventHandler>(&mut handler);
+    r.expect("Error!");
```
Comment (aduric) [inline bot/src/main.rs:18]: @mthjones How to get the underlying error out?

File: bot/src/event_handler.rs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+#[allow(unused_variables)]
+impl slack_api::EventHandler for EventHandler {
+    fn on_event(&mut self, cli: &mut slack_api::RtmClient, event: Result<&slack_api::Event, slack_api::Error>, raw_json: &str) {
+        println!("on_event(event: {:?}, raw_json: {:?})", event, raw_json);
+        
+        match event {
+            Ok(&Event::Message(Message::Standard { text: Some(ref t), .. } )) => {
+                match t.find(&format!("@{}", self.user)) {
+                    Some(_) => println!("You talkin' to me?"),
```
Comment (ekelundh) [inline bot/src/event_handler.rs:24]: We're not actually messaging back if pierre detects his name, right?

**Comment by ekelundh** [comment]
Where do we actually call daemon, anywhere?

File: bot/src/main.rs (hunk 4 above)
Comment (mthjones) [inline bot/src/main.rs:18]: With `.expect`, you can't get the underlying error. What you could do is use `if let`, like:

``` rust
if let Err(e) = cli.login_and_run::<EventHandler>(&mut handler) {
    println!("{}", e);
}
```

**Pull request merged**